File: api-evil-plotter-project/app/controllers/api/v1/notes_controller.rb
class Api::V1::NotesController < ApplicationController
  before_action :set_note, only: [:show, :update, :destroy]

  def index
    @notes = Note.all.where("board_id = ?", params[:board_id])
    render json: @notes
  end

  def show
    @note = Note.all.where("board_id = ? and id = ?", params[:board_id], params[:id])
    render json: @note
  end

  def create
    @note = Note.new(note_params)
    if @note.save
      render json: @note
    else
      puts "create error"
    end
  end

  def update
    @note.update(note_params)
  end

  def delete
    @note.destory
  end

  private

  def set_note
    @note = Note.find(params[:id])
  end

  def note_params
    params.require(:note).permit(:id, :title, :content, :left, :top, :height, :width)
  end

end

File: js-app-evil-plotter-group-project/src/controllers/notesController.js
class NotesController {
  constructor() {
    $('a#newNote').on('click.newNote', function(){
      const noteView = new NoteView
      const html = noteView.noteForm()
      const board = Board.all.forEach(function(board) {
        const board_id = $('.board-id').html()
        const user_id = sessionStorage.user_id
        if (board.id === board_id) {
          board.addNote("", "Title", "Content", '0', '0', '200px', '200px', board_id, user_id )
          // (id, title, content, left, top, height, width, board_id, user_id)
        }
      })





      // add to new board object
      // const board =  Board.all.find()

      // Note.all(board_id).then(function(data) {
      //   data.forEach(function(note) {
      //     Board.addNote('default title', 'type your note here', '0px', '0px', '200px', '200px')
      //   })
      // })
      $('div#noteContainer').html(board.notes)
    })

    $('a#save').on('click.save', () => this.save(new_board))
  }

  create() {
  }

  delete() {}

  update(){

  }
}
